index.d.ts

```diff
@@ -0,0 +1,39 @@
+// copied from Redux so that we don't have to add it as a dependency
+interface Action<T = any> {
+  type: T
+}
+
+export type Enhancement = object;
+
+export type EnhancerId = string | symbol;
+
+type ActionEnhancerById = {
+  readonly id: EnhancerId;
+};
+
+type ActionEnhancerByType = {
+  actionTypes: string[];
+};
+
+export type ActionEnhancer<S, E extends Enhancement, V = any> = (
+  | ActionEnhancerById
+  | ActionEnhancerByType) & {
+  mapState: (state: S, value: V) => E;
+};
+
+export type UnenhancedAction<T> = Action<T> & {
+  [index: string]: any;
```

Comment:
Technically this index signature should be `[index: string | symbol]: any`, however Symbols are not constant in TypeScript by default. You have to say `unique symbol` to tell TypeScript that the Symbol won't change and can be used as a signature.. But you can't use the `unique` keyword here. Also, you just generally can't use symbols as index signatures; only string and number are valid index signatures right now.

But even with that.. I get no compilation errors when I use a `Symbol` ID. I'm not entirely sure why, but if the compiler is happy with it I'm inclined to leave it be for now.

Reply:
weird yea let's leave it